### PR TITLE
MIDI-19: Maestro web streamlit release

### PR DESCRIPTION
**Pull request by roszcz**
Instead of generating mp3 on demand, I cached everything in Digital Ocean storage. The bucket is not public, so you need to set up credentials in order to run this (see [README](https://github.com/Nospoko/maestro-midi-streamlit/blob/a6d0c257842e650c6711571bcd96b22fac170f72/README.md)).

I also made it much faster by using a cached `pd.DataFrame` instead of huggingface dataset object to run filtering.

This branch is deployed to: https://maestro.pianoroll.ai/

### CODE REVIEW

File: dashboard.py [hunk 1]
```diff
@@ -1,25 +1,131 @@
...
+
+    # Get unique composer names from the dataset
+    composers = df.composer.unique()
+
+    # Create a combo box to select a composer
+    selected_composer = st.sidebar.selectbox("Select a composer:", composers)
+
+    # Create a text input to search by title
+    titles = df[df.composer == selected_composer].title.unique()
+    selected_title = st.sidebar.selectbox("Select title:", titles)
```
Comment (SamuelJanas) [inline dashboard.py:113]: This works great but my concern is whether it's not too limiting, currently it's only possible to search for titles inside the selected composer. 

Edit: wrong line

**Comment by SamuelJanas** [comment]
The comment I left was the only thing that caught my attention, apart from that everything looks good to me and works faster.

**Review by SamuelJanas — APPROVED** [review]
The comment I left was the only thing that caught my attention, apart from that everything looks good to me and works faster.

File: dashboard.py (hunk 1 above)
Comment (roszcz) [reply dashboard.py:113]: Yes, hopefully we'll be able to figure out a better UI for this soon :)

**Pull request merged**